app/src/main/java/com/example/tasky/feature_login/data/repository/UserRemoteRemoteRepositoryImpl.kt

```diff
@@ -24,54 +23,50 @@ class UserRemoteRemoteRepositoryImpl(
         println("hello from the repository. The appname is $appName")
     }
 
-    override suspend fun registerUser(registerUserInfo: RegisterUserInfo): Result<Unit, DataError.Network> {
+    override suspend fun registerUser(
+        name: String,
+        email: String,
+        password: String
+    ): Result<Unit, DataError.Network> {
         return try {
-            val response = api.registerUser(registerUserInfo = registerUserInfo)
+            val response =
+                api.registerUser(registerUserInfo = RegisterUserInfo(name, email, password))
             if (response.isSuccessful) {
                 Result.Success(Unit)
             } else {
-                val errorMessage =
-                    apiUtilFunctions.parseErrorResponse(response.errorBody()?.string())
                 val error = when (response.code()) {
-                    401 -> DataError.Network.Unauthorized(errorMessage)
-                    408 -> DataError.Network.RequestTimeout(errorMessage)
-                    413 -> DataError.Network.PayloadTooLarge(errorMessage)
-                    else -> DataError.Network.Unknown(errorMessage)
+                    401 -> DataError.Network.UNAUTHORIZED
+                    408 -> DataError.Network.REQUEST_TIMEOUT
+                    413 -> DataError.Network.PAYLOAD_TOO_LARGE
+                    else -> DataError.Network.UNKNOWN
                 }
-
                 Result.Error(error)
             }
-        } catch (e: HttpException) {
-            val error = apiUtilFunctions.getHttpErrorMessage(e)
-            Result.Error(error)
         } catch (e: IOException) {
```

Comment:
do I still need this catch? this part is unclear. maybe I can remove the whole try/catch block and just keep the if statement

Reply:
I think you do still need this, since it's not directly related to a response result, but about being able to make the request in the first place.. But I'm not 100% sure myself here, you can easily try this out by executing the request with airplane mode turned on. If you get the no internet error, you need this catch.

Reply:
I tried with wifi off and it needs the catch IO Exception block